lightning/src/ln/data_persister.rs

```diff
@@ -0,0 +1,27 @@
+//! Logic for persisting data from ChannelMonitors on-disk. Per-platform data
+//! persisters are separated into the lightning-persist-data crate.
+//! These objects mainly interface with the SimpleManyChannelMonitor when a
+//! channel monitor is added or updated, and when they are all synced on startup.
+use chain::channelmonitor::{ChannelMonitor, ChannelMonitorUpdate, ChannelMonitorUpdateErr};
+use chain::keysinterface::ChannelKeys;
+use chain::transaction::OutPoint;
+use std::collections::HashMap;
+
+/// ChannelDataPersister is responsible for persisting channel data: this could
+/// mean writing once to disk, and/or uploading to several backup services.
```

Comment:
I think we should document that how you choose to backup channel data has an impact on performance channel monitor update acceptance and thus on HTLC relay. So you may only send to memory, then between updates flush-to-disk.

Reply:
For full safety, independent on the goodwill of the peer, you have to persist before responding to the peer.  Otherwise you may revoke a tx, but crash before you persist that fact, and then be penalized when you come back up and broadcast a tx you shouldn't.

Also, might be good to discuss reaching consensus among backups, e.g. by majority vote.

Reply:
Exactly this for the documentation comment.

I think majority vote is okay for consensus among backup peers, as we assume they're the backend of channel-monitor and don't have signing/blockchain view/broadcast capabilities, one of them fallen behind can't propagate revoked state.

Reply:
This seems reasonable, as long as the majority also agree on the latest and highest update ID. Updated!